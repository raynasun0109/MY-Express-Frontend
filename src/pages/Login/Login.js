import * as React from 'react';
import Button from '@mui/material/Button';
import CssBaseline from '@mui/material/CssBaseline';
import TextField from '@mui/material/TextField';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import './Login.scss';

function Copyright(props) {
  return (
    <Typography variant="body2" color="text.secondary" align="center" {...props}>
      {'Copyright Â© '}
      <>
        MY Shopaholic
      </>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

export default function Login () {
    const handleSubmit = (event) => {
        event.preventDefault();
        const data = new FormData(event.currentTarget);
        console.log({
          email: data.get('email'),
          password: data.get('password'),
        });
      };

    return (
        <div className="login_container">
          <Container component="main" maxWidth="xs" className="login_container_content">
            <CssBaseline />
            <Box
              sx={{
              marginTop: 8,
              display: 'flex',
              flexDirection: 'column',
              alignItems: 'center',
              }}
            >
              <div className="login_container_content_login">
                Login
              </div>
              <Box component="form" onSubmit={handleSubmit} noValidate sx={{ mt: 1 }}>
                <TextField
                  margin="normal"
                  required
                  fullWidth
                  id="email"
                  label="Email Address"
                  name="email"
                  autoComplete="email"
                  autoFocus
                  className="login_input_field"
                />
                <TextField
                  margin="normal"
                  required
                  fullWidth
                  name="password"
                  label="Password"
                  type="password"
                  id="password"
                  autoComplete="current-password"
                  className="login_input_field"
                />
              <FormControlLabel
                control={<Checkbox value="remember" color="primary" />}
                label="Remember me"
              />
              <Button
                type="submit"
                fullWidth
                variant="contained"
                className="login_confirm_btn"
                sx={{ mt: 3, mb: 2 }}
              >
                Confirm
              </Button>
              <Grid container>
                <Grid item xs>
                  {/** TODO Forgot password */}
                  <Link href="#" variant="body2" className="signin_footer">
                    Forgot password?
                  </Link>
                </Grid>
                <Grid item>
                  <Link href="/register" variant="body2" className="signin_footer">
                    {"Don't have an account? Sign Up"}
                  </Link>
                </Grid>
              </Grid>
            </Box>
          </Box>
        <Copyright sx={{ mt: 8, mb: 4 }} />
      </Container>

        </div>
    );
};