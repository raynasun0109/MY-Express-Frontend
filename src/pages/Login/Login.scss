@import './../../config/_styles.scss';

.login_container{
    margin-top: 50px;
    margin-bottom: 100px;
    
    &_content{
        border:1px solid $base_color;
        border-radius: $border-radius_container;
        &_form {
            .MuiGrid-root{
                .MuiFormControl-root {
                    width: 100%;
                }
                .form_help_text{
                    color: $red!important;
                }
            }
        }
        &_login{
            @extend .big_font;
            color:$red;
            font-weight: $heavy_bold;
            margin-bottom: 20px;
        }
        .login_action_container{
            margin-bottom: 20px;
            margin-left: 16px;
        }
        .signin_footer{
            @extend .small_font;
            color:$red;
            margin-top: 10px;
        }
        .login_confirm_btn{
            background-color: $red;
            @extend .red_button;
        }
        .login_confirm_btn:hover{
            background-color: $red;
        }
        .login_input_field{
            .Mui-focused{
                color: black;
            }    
        }
    }
}

@media only screen and (max-width: 599px) {
    .login_container{
        .MuiContainer-root {
            max-width: 80%;
        }
    }
}

@media only screen and (max-width: 463px) {
    .login_action_container {
        display: flex;
        flex-direction: column!important;
        align-items: center;
        .MuiGrid-item {
            margin-bottom: 15px;
        }
    }
}