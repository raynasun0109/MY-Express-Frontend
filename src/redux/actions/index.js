export const removeCountry=(item)=>{
    return {
        type:"removeCountry",
        country:item
    }
}

export const addCountry=(item)=>{
    return {
        type:"addCountry",
        country:item
    }
}