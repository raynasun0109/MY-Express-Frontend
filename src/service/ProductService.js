import * as router from "./Routers";
