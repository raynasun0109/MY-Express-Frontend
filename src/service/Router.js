export const baseUrl="http://localhost:3001";

// products
export const allProducts="/products/all";

export const latestProducts="/products/latest";

export const getOneProduct="/products/get/one";

export const updateOneProduct="/products/update/one";

export const addOneProduct="/products/add/one";

export const getProductsFromMerchant="/products/get/own_by_merchant";

export const deleteOneProduct="/products/delete/one";


//users
export const getOneUser="/users/info/one";

export const registerOneUser="/users/register/one";

export const userLogin="/users/login";

export const updateOneUser="/users/update/one";

export const updateOneShoppingCart="/users/update/shoppingcart/one";

export const fetchOneShoppingCart="/users/get/shoppingcart/one";


//orders
export const addOneOrder="/orders/add/one";

export const getOrdersFromOneUser="/orders/get/all/oneUser";


//transaction
export const addOneTransaction="/transaction/add/one";

export const getOneTranscationFromOneOrder="/transaction/get/one/fromOneOrder";

export const getTranscationFromSameOrder="/transaction/get/one/fromSameOrder";

export const getTranscationFromSameMerchant="/transaction/get/one/fromSameMerchant";

export const updateOneTransaction="/transaction/update/one";
