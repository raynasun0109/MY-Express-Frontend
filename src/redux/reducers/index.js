// import localStorage from "localStorage";

// const initState={
//     countryList:JSON.parse(localStorage.getItem("learningList"))?JSON.parse(localStorage.getItem("learningList")).countryList:[],
//     totalNumber:JSON.parse(localStorage.getItem("learningList"))?JSON.parse(localStorage.getItem("learningList")).countryList.length:0,
// }
import products from './products';
import { combineReducers } from 'redux'
const reducer = combineReducers({
    products
  })
  
  export default reducer
