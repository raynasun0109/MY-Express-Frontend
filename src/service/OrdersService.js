import * as router from "./Router";
import axios from 'axios';

export function addOneOrder (params){
    return axios.post(`${router.baseUrl}${router.addOneOrder}`,params).then(res => {
        return res;
    });
}

export function getOrdersFromOneUser (params){
    return axios.post(`${router.baseUrl}${router.getOrdersFromOneUser}`,params).then(res => {
        return res;
    });
}