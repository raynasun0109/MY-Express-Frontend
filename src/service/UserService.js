import * as router from "./Router";
import axios from 'axios';

export function getOneUser (params){
    return axios.post(`${router.baseUrl}${router.getOneUser}`,params).then(res => {
        return res;
    });
}

export function registerOneUser (params){
    return axios.post(`${router.baseUrl}${router.registerOneUser}`,params).then(res => {
        return res;
    });
}

export function userLogin (params){
    return axios.post(`${router.baseUrl}${router.userLogin}`,params).then(res => {
        return res;
    });
}