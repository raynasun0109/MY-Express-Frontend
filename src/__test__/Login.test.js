import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react';
import Login from '../pages/Login/Login';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {MemoryRouter as Router} from 'react-router-dom';
import { userLogin} from "../service/UserService";
import rootReducer from '../redux/reducers';
import {createStore} from 'redux';

const mockStore = configureStore([]);
const store = createStore(rootReducer);

jest.mock('universal-cookie', () => {
    return jest.fn().mockImplementation(() => {
        return {
            set: jest.fn(),
            get: jest.fn(),
        };
    });
});

jest.mock('../service/UserService', () => {
    return {
        userLogin: jest.fn().mockImplementation(() => {
            return Promise.resolve({
                data: {
                    code: 1,
                    data: [
                        {
                            email: 'test@test.com',
                            shopping_cart: [],
                        },
                    ],
                },
            });
        }),
    };
});

describe('Login component', () => {
    const originalScrollTo = window.scrollTo;
    let store;

    beforeEach(() => {
        window.scrollTo = jest.fn();
        store = mockStore({
            shoppingCart: [],
        });
    });


    afterEach(() => {
        window.scrollTo = originalScrollTo;

        jest.clearAllMocks();
    });

    it('should render correctly', () => {
        const {asFragment} = render(
            <Router>
                <Provider store={store}>
                    <Login/>
                </Provider>
            </Router>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should submit form correctly', async () => {
        userLogin.mockImplementationOnce(() => Promise.resolve({
            "code": 1,
            "msg": "Login Successfully",
            "data": [
                {
                    "uuid": "f92ecb0e-e2f9-4d6e-93ff-20bddcad5493",
                    "first_name": "undefined",
                    "password": "12121",
                    "type": "undefined",
                    "created_at": "1682066832949",
                    "update_at": "1682066832949",
                    "email": "john.doe@example.com",
                    "last_name": "undefined",
                    "shopping_cart": "[]"
                }
            ]
        }));
        const { getByText, getByLabelText} = render(
            <Router>
                <Provider store={store}>
                    <Login/>
                </Provider>
            </Router>
        );

        try{
            fireEvent.change(getByLabelText('Email Address'), {
                target: {value: 'test@test.com'},
            });
            fireEvent.change(getByLabelText('Password'), {
                target: {value: 'test'},
            });
    
    
            fireEvent.click(getByText('Confirm'));
        } catch(e){
            console.log(e);
            
        }
        

        // await waitFor(() => expect(userLogin).toHaveBeenCalledTimes(1));
    });
});
