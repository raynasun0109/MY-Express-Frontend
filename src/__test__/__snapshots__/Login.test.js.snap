// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="login_container"
  >
    <main
      class="MuiContainer-root MuiContainer-maxWidthXs login_container_content css-nzinn5-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-binzgt"
      >
        <div
          class="login_container_content_login"
        >
          Login
        </div>
        <form
          class="MuiBox-root css-164r41r"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 login_container_content_form css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root login_input_field css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="email"
                    id="email-label"
                  >
                    Email Address
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="email"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="email"
                      name="email"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Email Address
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <p
                class="MuiFormHelperText-root form_help_text css-1d1r5q-MuiFormHelperText-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root login_input_field css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="password"
                    id="password-label"
                  >
                    Password
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="current-password"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="password"
                      name="password"
                      type="password"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-14lo706"
                      >
                        <span>
                          Password
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <p
                class="MuiFormHelperText-root form_help_text form_help_text_psw css-1d1r5q-MuiFormHelperText-root"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
              >
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    type="checkbox"
                    value="remember"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                >
                  Remember me
                </span>
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth login_confirm_btn css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Confirm
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container login_action_container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways signin_footer css-101ca9i-MuiTypography-root-MuiLink-root"
                  href="#"
                >
                  Forgot password?
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways signin_footer css-101ca9i-MuiTypography-root-MuiLink-root"
                  href="/register"
                >
                  Don't have an account? Sign Up
                </a>
              </div>
            </div>
          </div>
        </form>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
